src/optimal_image/compress/image.rs

```diff
@@ -32,8 +37,18 @@ impl<'a> VipsImage<'a> {
         path: S,
     ) -> Result<VipsImage<'a>, Box<Error>> {
         let path = CString::new(path)?;
-        let c =
-            unsafe { vips::vips_image_new_from_file(path.as_ptr(), NULL_LIST) };
+        let c = unsafe {
+            let mut out: *mut *mut vips::VipsImage =
+                vips::vips_image_new() as *mut *mut vips::VipsImage;
+            let rgb_image =
+                vips::vips_image_new_from_file(path.as_ptr(), NULL_LIST);
+            // new_from_file can return RGB image, we have to add alpha in order
+            // to be sure to have an RGBA image
+            vips::vips_addalpha(rgb_image, out);
```

Comment:
well, image_new_from_file correctly set the alpha only if it is in the file. an alternative is using addbands(255), but I think it would do exactly the same thing.